packages/chatbot-server-mongodb-public/src/eval/fuzzyLinkMatch.ts

```diff
@@ -1,20 +1,40 @@
 /**
-  Performs a case-insensitive, partial match
-  that the `expected` link or link fragment
-  ends with the `actual` link or link fragment.
-  Removes trailing `/` from paths.
-  Checks based solely on path, ignoring domain, query, and fragment.
+  Performs a case-insensitive match between two URLs or URL fragments.
+  
+  First attempts to match based on paths:
+  - Removes trailing slashes
+  - Checks if actual path ends with expected path (ignoring domain, query, and fragment)
+  
+  If either path is empty/invalid, falls back to exact match of normalized URLs:
+  - Removes protocol (http/https)
+  - Removes 'www.' prefix
+  
+  @param expected - The expected URL or URL fragment
+  @param actual - The actual URL or URL fragment to compare against
+  @returns true if URLs match according to above rules, false otherwise
+ */
+export function fuzzyLinkMatch(expected: string, actual: string) {
+  const cleanActualPath = getCleanPath(actual);
+  const cleanExpectedPath = getCleanPath(expected);
 
-  If either input is not a valid URL,
-  it treats the input as a plain string
-  and performs a substring match.
+  // if cleaned path is not an empty string, compare cleaned paths
+  if (cleanActualPath && cleanExpectedPath) {
+    return cleanActualPath.endsWith(cleanExpectedPath);
+  } else {
+    // compare full normalized paths
```

Comment:
```suggestion
    // compare normalized full URLs
```
I think this is more accurate? Disregard if I'm wrong.